### PR TITLE
[should be deleted] ND: people scraper rewritten for new website

### PR DESCRIPTION
**Pull request by chrisyamas**
[Updated description on 11/24/22]: This PR should be deleted because the new rewrite of the ND people scraper was accidentally pushed from a different branch into the same pull request that has the rewrite of the ND events scraper. [That PR can be found here](https://github.com/openstates/openstates-scrapers/pull/4162).

[Original description]: Entire rewrite of ND people scraper built for state's new website. Includes pagination.

### CODE REVIEW

File: scrapers_next/nd/people.py [hunk 1]
```diff
@@ -1,118 +1,160 @@
...
+        start_date = ""
+        if status:
+            status_text = status[0].text_content().strip()
+            if "Expelled" in status_text:
+                self.skip()
+            elif "Active" in status_text:
+                start_date = status_text.split()[-1].strip()
+
+        district_div = item.find_class("district")[0].text_content().strip()
+        dist_match = re.search(r"District\s+\S+", district_div)
```
Comment (NewAgeAirbender) [inline scrapers_next/nd/people.py:132]: Should try to capture the district number in the search using something like [this](https://github.com/openstates/openstates-scrapers/blob/f97fe365665733a343e1addbf4f19ef25e5417bf/scrapers_next/tx/people.py#L86) so you don't need the next few lines

File: scrapers_next/nd/people.py [hunk 2]
```diff
@@ -1,118 +1,160 @@
...
+        for phone_type in ("Cell", "Home"):
+            if collected_contacts.get(phone_type):
+                p.extras[phone_type.lower()] = collected_contacts[phone_type]
+
+        p.add_source(str(self.source), "member details page")
+
+        members_list_url = str(self.input.url)
+        p.add_source(members_list_url, "members list page")
+
+        url_match = re.search(r".+members", members_list_url)
```
Comment (NewAgeAirbender) [inline scrapers_next/nd/people.py:78]: Just one list page and the member detail page would be enough, but if you want to have it be the cleaner looking first page url, go for it!

**Review by NewAgeAirbender — CHANGES_REQUESTED** [review]
Nice work, had a few areas to tweak, but I think the contact pattern check works well here.

File: scrapers_next/nd/people.py (hunk 1 above)
Comment (chrisyamas) [reply scrapers_next/nd/people.py:132]: I implemented [this one-line solution](https://github.com/openstates/openstates-scrapers/pull/4162/commits/ed81cdcef99f666fd2d8464c036fe5263b687dee#diff-b4eb1b6b6189a33b171b390670cb2f7c25db0b25acd7e596c8c5afb685d778a6R136) in the most updated commit (which is on nd_events branch).

File: scrapers_next/nd/people.py (hunk 2 above)
Comment (chrisyamas) [reply scrapers_next/nd/people.py:78]: I took your guidance and [deleted those lines](https://github.com/openstates/openstates-scrapers/pull/4162/commits/ed81cdcef99f666fd2d8464c036fe5263b687dee#diff-b4eb1b6b6189a33b171b390670cb2f7c25db0b25acd7e596c8c5afb685d778a6L78-L81) in the new commit.

**Comment by chrisyamas** [comment]
> Nice work, had a few areas to tweak, but I think the contact pattern check works well here.

Thanks! Those areas got tweaked in the newest commit which is now a part of this [Nd events and people scraper rewrites](https://github.com/openstates/openstates-scrapers/pull/4162) pull request. I had accidentally pushed up recent additional tweaks to the ND people scraper while still on my nd_events I had been working on. However since that PR has the most updated version of both the new ND events and ND people scrapers, I think it would makes sense to merge it into openstates:main and then close/delete this branch and PR without merging as it's behind the other branch/PR at this point.

**Pull request closed without merging**